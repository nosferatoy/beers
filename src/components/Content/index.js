import React from 'react';
import cx from 'classnames';
import './index.scss';

const Content = ({
	children,
	vcenter = false,
	hcenter = false,
	padded = false,
	noHeader = false
}) => (
	<main
		className={cx('content', {
			'content--vcenter': vcenter,
			'content--hcenter': hcenter,
			'content--padded': padded,
			'content--noHeader': noHeader
		})}
	>
		{children}
	</main>
);

export default Content;
