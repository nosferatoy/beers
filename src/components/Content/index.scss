.content {
	--header-size: calc(var(--header-height) + 1rem);

	margin: var(--header-size) auto 0 auto;
	min-width: 320px;
	max-width: 800px;

	&--noHeader {
		--header-size: 0px;
	}

	&--vcenter {
		text-align: center;
	}
	&--hcenter {
		display: flex;
		align-items: center;
		flex-wrap: wrap;
		justify-content: center;
		> * {
			flex: 1;
		}
	}

	&--padded {
		padding: 1rem;
	}
}
