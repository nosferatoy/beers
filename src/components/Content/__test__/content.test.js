import React from 'react';
import { render } from '@testing-library/react';
import Content from '../index';

describe('Test Content component', () => {
	it('Render Content', () => {
		const { container } = render(
			<Content>
				<div>Test 1</div>
				<div>Test 2</div>
				<div>Test 3</div>
				<div>Test 4</div>
			</Content>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render Content horizontaly', () => {
		const { container } = render(
			<Content hcenter={true}>
				<div>Test 1</div>
				<div>Test 2</div>
				<div>Test 3</div>
				<div>Test 4</div>
			</Content>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render Content verticaly', () => {
		const { container } = render(
			<Content vcenter={true}>
				<div>Test 1</div>
				<div>Test 2</div>
				<div>Test 3</div>
				<div>Test 4</div>
			</Content>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render Content padded', () => {
		const { container } = render(
			<Content padded>
				<div>Test 1</div>
				<div>Test 2</div>
				<div>Test 3</div>
				<div>Test 4</div>
			</Content>
		);
		expect(container).toMatchSnapshot();
	});

	it('Render Content no header', () => {
		const { container } = render(
			<Content noHeader>
				<div>Test 1</div>
				<div>Test 2</div>
				<div>Test 3</div>
				<div>Test 4</div>
			</Content>
		);
		expect(container).toMatchSnapshot();
	});
});
