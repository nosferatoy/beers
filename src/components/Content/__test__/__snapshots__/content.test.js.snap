// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Content component Render Content 1`] = `
<div>
  <main
    class="content"
  >
    <div>
      Test 1
    </div>
    <div>
      Test 2
    </div>
    <div>
      Test 3
    </div>
    <div>
      Test 4
    </div>
  </main>
</div>
`;

exports[`Test Content component Render Content horizontaly 1`] = `
<div>
  <main
    class="content content--hcenter"
  >
    <div>
      Test 1
    </div>
    <div>
      Test 2
    </div>
    <div>
      Test 3
    </div>
    <div>
      Test 4
    </div>
  </main>
</div>
`;

exports[`Test Content component Render Content no header 1`] = `
<div>
  <main
    class="content content--noHeader"
  >
    <div>
      Test 1
    </div>
    <div>
      Test 2
    </div>
    <div>
      Test 3
    </div>
    <div>
      Test 4
    </div>
  </main>
</div>
`;

exports[`Test Content component Render Content padded 1`] = `
<div>
  <main
    class="content content--padded"
  >
    <div>
      Test 1
    </div>
    <div>
      Test 2
    </div>
    <div>
      Test 3
    </div>
    <div>
      Test 4
    </div>
  </main>
</div>
`;

exports[`Test Content component Render Content verticaly 1`] = `
<div>
  <main
    class="content content--vcenter"
  >
    <div>
      Test 1
    </div>
    <div>
      Test 2
    </div>
    <div>
      Test 3
    </div>
    <div>
      Test 4
    </div>
  </main>
</div>
`;
